### PR TITLE
docs: add deployment guide

### PR DESCRIPTION
**Pull request by hongalex**
**What type of PR is this?**
> /kind docs

**What this PR does / Why we need it**:
Adds quick deployment guide for setting up Open Saves

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/deploying.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+### Check Datastore
+
+In your browser, navigate to the [Datastore dashboard](https://console.cloud.google.com/datastore).
+You should see several entities here, from running the example code. Try looking for different
+kinds in the search bar at the top, specifically "Store", "Record", "Blob".
+
+### Check Memorystore
+
+While you can't see individual keys directly, navigate to the [Memorystore dashboard](https://console.cloud.google.com/memorystore)
```
Comment (irataxy) [inline docs/deploying.md:210]: missing period

File: docs/deploying.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+Alternatively, you can use `gsutil` to list the object via command line.
+
+```bash
+gsutil ls $BUCKET_PATH
+```
+
+## Next steps
+
+You have successfully deployed an Open Saves server on Cloud Run and saved and read data from it.
```
Comment (irataxy) [inline docs/deploying.md:228]: You have successfully deployed an Open Saves server on Cloud Run, saved data to it, and then read that data back from it.

File: docs/deploying.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+### Check Datastore
+
+In your browser, navigate to the [Datastore dashboard](https://console.cloud.google.com/datastore).
+You should see several entities here, from running the example code. Try looking for different
+kinds in the search bar at the top, specifically "Store", "Record", "Blob".
+
+### Check Memorystore
+
+While you can't see individual keys directly, navigate to the [Memorystore dashboard](https://console.cloud.google.com/memorystore)
+Select the instance that you created, and then select the "Keys in database" graph.
```
Comment (irataxy) [inline docs/deploying.md:211]: My keys didn't show up right away. Maybe note this and the number of keys you should expect to see.

File: docs/deploying.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+export VPC_CONNECTOR=open-saves-vpc
+```
+
+### Starting the cache store
+
+Run the following command to create a Redis instance by using [Memorystore](https://cloud.google.com/memorystore). This
+will give you an instance with a private IP, which you pass into Open Saves.
+
+```bash
+gcloud redis instances create --region=$GCP_REGION $REDIS_ID
```
[the remark this answers is not in the document]
Comment (hongalex) [reply docs/deploying.md:75]: I added some instructions for that at the top, under "Before you begin".

File: docs/deploying.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+Next, enable the Serverless VPC Access API for your project.
+
+```bash
+gcloud services enable vpcaccess.googleapis.com
+```
+
+Create the VPC connector. This step may take some time.
+
+```bash
+gcloud compute networks vpc-access connectors create $VPC_CONNECTOR \
```
[the remark this answers is not in the document]
Comment (hongalex) [reply docs/deploying.md:120]: Ah, good catch! Changing.

File: docs/deploying.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+```
+
+The output should contain the line state: READY.
+
+### Cloud Firestore in Datastore mode
+
+Follow the [quickstart guide](https://cloud.google.com/datastore/docs/quickstart)
+to set up a database in **Datastore** mode.
+
+![datastore_mode](images/datastore.png)
```
[the remark this answers is not in the document]
Comment (hongalex) [reply docs/deploying.md:139]: This was requested by Mark and Jon when they were attempting this. It's probably obvious for most users, but considering this operation can't be undone, we want to be as explicit as possible.

File: docs/deploying.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+ENDPOINT=${ENDPOINT#https://} && echo ${ENDPOINT}
+```
+
+Next, run the example client application to make sure everything worked.
+Make sure you have Go v1.14 or later installed.
+
+```bash
+git clone https://github.com/googleforgames/open-saves
+go run examples/grpc-all/main.go -address=$ENDPOINT:443
```
[the remark this answers is not in the document]
Comment (hongalex) [reply docs/deploying.md]: Yes, thanks! Fixed.

File: docs/deploying.md (hunk 3 above)
Comment (hongalex) [reply docs/deploying.md]: Added clarifiation.

Also, I noted you should see "several". This example might change in the future so I don't want to fix this to a specific number of keys.

**Pull request merged**